### DISCUSSION TITLE
How can I configure a snapshot policy with the operator?

### DISCUSSION THREAD

**Discussion by nilushancosta** [Opster/opensearch-k8s-operator#796] [Q&A] [answered]
I want to automate snapshots by creating a [snapshot policy](https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/snapshots/sm-api/#create-or-update-a-policy). Instead of making an API call to create it, is there a way to create it using a configuration or a custom resource of the operator?
Similar to how we could use an `OpenSearchISMPolicy` to create an Index State Management Policy or the [snapshotRepositories](https://github.com/opensearch-project/opensearch-k8s-operator/blob/main/docs/userguide/main.md#configuring-snapshot-repositories) configuration to register a snapshot repository instead of making API calls to OpenSearch.

**Comment by swoehrl-mw** [comment, marked as answer]
Hi @nilushancosta. Currently there is no CR or other support from the operator to create snapshot policies. But this is a good thing to have. I'll create a feature request issue for it.

**Reply by nilushancosta** [reply]
Thanks for the clarification @swoehrl-mw

**Reply by nilushancosta** [reply]
GitHub issue discussing this feature - https://github.com/opensearch-project/opensearch-k8s-operator/issues/806
